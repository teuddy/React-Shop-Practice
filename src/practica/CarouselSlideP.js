/*

    <figure class="" style="">
        <img src="/images/234343..png">        
    <figure/>


<div/>

*/

/*
CSS
    .figure
    width: 100%;
    height: 300px;
    display:flex

    
*/

import React from 'react';
import PropTypes from 'prop-types';

const figureCss = {
  width: '95%',
  height: '300px',
};
const imgCss = {
  objectFit: 'cover',
  width: '100%',
  height: '100%',
};

const CarouselSlideP = ({ imgUrl }) => (
  <figure style={figureCss}>
    <img src={imgUrl} style={imgCss} />
  </figure>
);

CarouselSlideP.propTypes = {
  imgUrl: PropTypes.string,
};

export default CarouselSlideP;
