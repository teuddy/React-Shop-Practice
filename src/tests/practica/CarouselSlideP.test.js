/*
1 - El componente es representado por un elemento figure

2 - <figure/> tiene un property style, con los valores ""

3 - <figure> tiene un solo hijo, 
    (3.5) El cual es un elemento de tipo img

4 - El elemento <img> tiene una propiedad src con el valor
    que le pasamos
*/

import React from 'react';
import { shallow } from 'enzyme';
import CarouselSlideP from '../../practica/CarouselSlideP';

describe('<CarouselSlideP/>', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<CarouselSlideP />);
  });

  it('no cambia', () => {
    wrapper.setProps({
      imgUrl:
        'https://4.img-dpreview.com/files/p/E~TS590x0~articles/3925134721/0266554465.jpeg',
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('Es un figure', () => {
    expect(wrapper.type()).toEqual('figure');
  });
  it('Figure tiene un prop: {style: "valor"}', () => {
    const figureCss = {
      width: '95%',
      height: '300px',
    };
    expect(wrapper.prop('style')).toEqual(figureCss);
  });
  it('el elemento figure tiene un solo hijo el cual es un img', () => {
    expect(wrapper.find('figure').children()).toHaveLength(1);
    expect(wrapper.find('figure').childAt(0).type()).toBe('img');
  });
  it('El elemento img tiene estos estilos css', () => {
    const imgCss = {
      objectFit: 'cover',
      width: '100%',
      height: '100%',
    };
    expect(wrapper.find({ style: imgCss }).type()).toBe('img');
  });
  it('elemento img tiene un prop: {src: "valor"', () => {
    const srcAttributeContent =
      'https://4.img-dpreview.com/files/p/E~TS590x0~articles/3925134721/0266554465.jpeg';
    wrapper.setProps({ imgUrl: srcAttributeContent });
    expect(wrapper.find('img').prop('src')).toBe(srcAttributeContent);
  });
});
