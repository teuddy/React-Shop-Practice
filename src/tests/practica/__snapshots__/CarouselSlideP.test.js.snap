// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarouselSlideP/> no cambia 1`] = `
<figure
  style={
    Object {
      "height": "300px",
      "width": "95%",
    }
  }
>
  <img
    src="https://4.img-dpreview.com/files/p/E~TS590x0~articles/3925134721/0266554465.jpeg"
    style={
      Object {
        "height": "100%",
        "objectFit": "cover",
        "width": "100%",
      }
    }
  />
</figure>
`;
